roles/common/tasks/hostname.yml

```diff
@@ -0,0 +1,59 @@
+---
+# roles/hostname/tasks/main.yml
+# The _domain variable functions as a FQDN to handle deployments using a subdomain, top level domain and www subdomain, or default rDNS. 
+# When domain is defined
+- name: set dns fact for subdomain
+  set_fact:
+    _domain: '{{ subdomain }}.{{ domain }}'
+    cacheable: yes
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+  when:
+    - domain is defined
+    - subdomain != "www"
+    
+- name: set dns fact for domain
+  set_fact:
+    _domain: '{{ domain }}'
+    cacheable: yes
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+  when: 
+    - domain is defined
+    - subdomain == "www"
+  
+- name: setting system hostname on master
+  hostname:
+    name: "{{ _domain }}" 
+  delegate_to: "{{ groups['spark_master'][0] }}"
+  run_once: true
+  when: 
+    - domain is defined
+
+# Since worker UIs are viewed through reverse proxy, hostname will always be default dns on workers
+- name: setting system hostname on workers
+  hostname:
+    name: "{{ ansible_default_ipv4.address | replace('.', '-') }}.ip.linodeusercontent.com"
+  when: 
+    - domain is defined
+    - "'spark_workers' in group_names"
```

Comment:
We should be using `_domain` to set the domain for the app. We define what the hostname is in the provision.yml and should use it. We can iterate through that data structure to set the hostname.

Reply:
Changes made to hostname related tasks.